### PR TITLE
Get user preferred languages on apple devices

### PR DESCRIPTION
**Pull request by ephphatha**
Adapts the sample code provided by @bubio in #3641 so GetLocales() returns a list of POSIX language codes on apple devices (hopefully... I don't have an environment set up to cross-compile this, lets see if the build works).

Co-authored-by: Bubio <bubio66@gmail.com>

### CODE REVIEW

File: Source/platform/locale.cpp [hunk 1]
```diff
@@ -15,13 +15,42 @@
...
+
+std::string IetfToPosix(string_view langCode)
+{
+	/*
+	 * Handle special case for simplified/traditional Chinese. IETF/BCP-47 specifies that only the script should be
+	 *  used to discriminate languages when the region doesn't add additional value. For chinese scripts zh-Hans is
+	 *  preferred over zh-Hans-CN (but platforms may include the region identifier anyway). POSIX locales don't use
+	 *  script in the same way so we need to convert these back to lang_region.
+	 */
+	if (langCode.substr(0, 7) == "zh-Hans") {
```
Comment (AJenbo) [inline Source/platform/locale.cpp:38]: starts_with

Comment (ephphatha) [reply Source/platform/locale.cpp:38]: That was introduced in C++20 unfortunately, doesn't look like we have a polyfill

**Review by AJenbo — DISMISSED** [review]

**Review by AJenbo — APPROVED** [review]

**Comment by bubio** [comment]
If locales is
    "en-JP",
    "ja-JP",
then the selected one is
   "ja-JP"
is the correct answer?

**Comment by AJenbo** [comment]
No, it should result in ja_JP

**Comment by ephphatha** [comment]
If you run with verbose logging active, what output do you get for the log message in options.cpp:935? (The line "Found {} user preferred locales")

**Comment by bubio** [comment]
> If you run with verbose logging active, what output do you get for the log message in options.cpp:935? (The line "Found {} user preferred locales")

I don't know how to do it.

**Comment by ephphatha** [comment]
Ahh, ok. With the order of configured languages on your system English/US English is prioritised below Japanese (and Korean... sorta). Is this your usual language order preference? If you got rid of the English (US) and English (Australia) entries you'd end up with the neutral english fallback happening almost immediately.

The search order is:
1. English (Japan) - not found, no fallback as a different english variant is provided further down the list.
2. Japanese (Japan) - redundant region specifier, not found with this tag.
3. Japanese - "neutral" fallback code, found. *search stops here*
4. Korean (Japan) - not found
5. Korean - "neutral" fallback code, this would not find a match as the provided translation is for Korean (Korea)
6. English - this was English (US), was transformed by the special case handling for the base language. This would be the language selected if step 3 wasn't a match.
7. Portuguese (Brazil) - this would be selected if both step 3 and 6 weren't matches.
8. Portuguese - "neutral" fallback code, not found
9. Chinese (Simplified) - next match if 3,6,7 failed
10. English (Australia) - not found
11. English - "neutral" fallback, duplicate of 6 because of the special handling for English (US).
12. Chinese (Traditional) - next match if 3,6,7,9 failed
13. Chinese - "neutral" fallback

**Comment by bubio** [comment]
Thank you very much.
I understand now.
 macOS 12
 iPad OS 15.2
I have confirmed that it works fine.


Usually, it's just ja_JP :)
For testing purposes, I set it up.

**Pull request merged**